src/run-codemod-on-file.ts

```diff
@@ -138,20 +138,21 @@ export default async function runCodemodOnFile(
       parse(source: string, opts: Record<string, unknown>) {
         const babelOpts = {
           ...getBabelOpts(),
-          ..._.pick(codemod, 'presets'),
+          ..._.pick(codemod, 'babelTransformOptions'),
```

Comment:
It seems like `getBabelOpts` should have higher precedence than `babelTransformOptions`, right? If the user passes `{ast: false}`, it'll break the codemod runner.

Generally, I'm a bit concerned this feature will be tricky to use, because it's hard for the user to know what they can safely pass, and what will interfere with what the codemod runner needs. But I'm ok just saying in the docs that this is "advanced", and the only thing you can safely pass is "presets". :smile:

What do you think?

Reply:
nm, I see you basically responded to this same question [earlier](https://github.com/NickHeiner/jscodemod/issues/28#issuecomment-923047678).